devito/operator/profiling.py

```diff
@@ -330,9 +339,36 @@ def trackable_subsections(self):
 
 class AdvancedProfilerVerbose2(AdvancedProfilerVerbose):
 
+    def instrument(self, iet, timer):
+        """
+        Instrument the given IET for C-level performance profiling.
+        """
+        iet = self.instrument_sections(iet, timer)
+        iet = self.instrument_ecalls(iet, timer)
+
+        return iet
+
+    def instrument_ecalls(self, iet, timer):
+        """
+        Instrument the given IET's elemental_calls for C-level performance profiling.
+        """
+        # Search for ElementalCalls *not* nested within other timer fields
+        if iet.name not in timer.fields:
+            elemental_calls = FindNodes(ElementalCall).visit(iet)
+            if elemental_calls:
+                mapper = {}
+                for i in elemental_calls:
+                    n = i.name
+                    assert n in timer.fields
+                    mapper[i] = i._rebuild(TimedList(timer=timer, lname=n, body=i))
+
```

Comment:
Aside from the fact that it's much simpler if

```
mapper = {i: i._rebuild(TimedList(timer=timer, lname=i.name, body=i)
                  for i in elemental_calls}
```

This PR is not doing what the description says. It's doing something much more general -- any unnested ElementalFunction would now be wrapped within timers. 

However, the PR description claims that this is only the case for MPI's compute0

Hence, the proper way to go, in my opinion, is to make `compute0` special -- akin to https://github.com/devitocodes/devito/blob/master/devito/mpi/routines.py#L1026